src/storage/checkpoint_io.cpp

```diff
@@ -0,0 +1,60 @@
+#include "storage/checkpoint_io.h"
+#include <vector>
+namespace terrier::storage {
+
+void BufferedTupleWriter::SerializeTuple(ProjectedColumns::RowView *row, ProjectedRow *row_buffer,
```

Comment:
Here you guys are trying to materialize every single tuple in the table. It is basically reimplementing the SqlTable::Select code. So at this point why not just grab the SlotIterator::SqlTable and dereference that to get a tuple slot and then call SqlTable::Select on that tuple slot. The SqlTable::Select will materialize the ProjectedRow you want and you can use that here, instead of having to reconstruct the entire ProjectedRow from scratch after already calling Scan.

Reply:
Also this isn't the purpose for how RowView was envisioned to be used. It is for use inside of the storage layer as a way of templating between ProjectedRows and ProjectedColumns. Ideally you guys should be needing to access the BlockLayout (that shouldn't be exposed outside of the storage layer) to get the RowView.

Reply:
After some discussion it looks like other people are using ProjectedColumns for purposes like you guys as well (https://github.com/cmu-db/terrier/pull/260#discussion_r274444563). So by using Scan you wouldn't have to reason about the visibility of the tuples since the SqlTable would handle that so it would be an easier process. To ensure that you guys don't need the BlockLayout we will try to work on being able to generate the RowView from a ProjectedColumn without needing BlockLayout

Reply:
We abandoned Scan and switched to Select in the latest version, and we do not require BlockLayout anymore. The problem becomes how can we get the schema of a table because we need the Oids of all its columns. Since there are no catalogs yet, is it reasonable to add an API in the SqlTable to return its schema?

Reply:
I would recommend against this because it would lead to more work in the end.  Instead, I would recommend either cherry-picking the API/files you need from the Catalog PR or laying out the API you need from the catalog and mocking it out in your test fixture so that it always provides some hard-coded or programmatically generated schema.